docs/proposals/config_data.md

```diff
@@ -0,0 +1,262 @@
+    "kind": "ConfigData",
+    "data": {
+        "config.yaml": "CONFIG_YAML_STRING_CONTENTS",
+        "spec.yaml": "SPEC_YAML_STRING_CONTENTS"
+    }
+}
+```
+``PodConfiguration2`` is intended to be used as a mounted volume containing two files with their respective contents.
+
+Then, configuration objects are persisted to the API server as follows:
+```bash
+kubectl create -f pod-configuration-1.json 
+kubectl create -f pod-configuration-2.json
+```
+
+We create a replication controller specification that creates pods using these two configuration objects:
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: dummy-app
+  template:
+    metadata:
+      labels:
+        app: dummy-app
+    spec:
+      # New pods will be created using the ConfigData objects.
+      # Each config object will be updated be it a volume or environment variables.
+      # But, some of them might require a restart on update.
+      # Note: If multiple env ConfigData objects are specified, let's call them Env1 and Env2.
+      #       If both of them contains a same named environment variable, Env2 will be the final one.
+      #       If Env1 is updated, the reload process will be Env1 first, then Env2.
+      #       Even though Env2 is not updated, the 'env' ConfigData objects are reloaded in the
+      #       order they are specified to prevent Env1 from overriding any of the variables in Env2.
+      # TODO: 'config-data' is a separate field because there are different ways to consume the object
+      #       and 'volume' might be a bit too specific. More discussions on this will be needed.
+      config-data:
+          env:
+          # PodConfiguration1 is loaded as environment variables into the
+          # main container of the pod. It requires a restart when the object is updated
+          - name: PodConfiguration1
+            on_update: restart
+          volume:
+          # PodConfiguration2 is mounted as a new volume to the main container of the pod.
+          - name: PodConfiguration2
+      containers:
+      - name: dummy-app
+        image: dummy-app
+        ports:
+        - containerPort: 80
+```
+
+example-replication-controller starts 2 replicas that runs the `dummy-app` container.
+Pod template states how these pods will make use of the ``ConfigData`` objects.
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ExampleReplicationController will use the ``PodConfiguration`` objects when starting/replicating pods.
+
+When the ``PodConfiguration`` objects are modified the new object will be retrieved and loaded into the component.
+
+### Considerations
+
+* ``ConfigData`` object might easily be extended to include other configuration components used in ``Kubernetes`` to configure it's behaviour one way or another (e.g. config files).
```

Comment:
Initially we envisioned ``ConfigData`` to be used with a few components. But, it evolved since then. It's best to remove this line to avoid confusion.